### DISCUSSION TITLE
How to install one file before start windows service

### DISCUSSION THREAD

**Discussion by zlishaojiez** [wixtoolset/issues#6761] [Questions]
I make one windows service installer, the windows service will call another exe,
I should install the exe file and the windows service exe together。
How can I make sure the exe which is called by windows service installed before start the windows service.
e.g.

     <Component Id="hst_outpost">
	     <File Id="hst_outpost.exe" Name="hst-outpost.exe" Source="../hst-outpost.exe" Vital="yes" KeyPath="yes"/>
     </Component>

     <Component Id="hst_service">				
	  <File Id="hst_service.exe" Name="hst-service.exe" DiskId="1" Source="$(var.hst-service.TargetPath)" Vital="yes" KeyPath="yes" 
         />
	<ServiceInstall Id="ServiceInstaller" Type="ownProcess" Vital="yes"
						Name="HstService" DisplayName=" Synthetic Testing"
						Description=" Synthetic Testing" Start="auto"
						Account="NT AUTHORITY\NetworkService" ErrorControl="normal" Interactive="no" />

	<ServiceControl Id="StartService"  Stop="both" Start="install" Remove="uninstall"
						Name="HstService" Wait="yes" />
     </Component>   

The service HstService will call hst-outpost.exe, how it can make sure hst-outpost.exe has installed before start HstService

**Comment by robmen** [comment]
The standard order of operations in the Windows Installer will install files before starting services. Should work as you desire.
